packages/core/data-transfer/lib/providers/local-file-destination-provider.ts

```diff
@@ -38,16 +38,26 @@ class LocalFileDestinationProvider implements IDestinationProvider {
   name: string = 'destination::local-file';
   type: ProviderType = 'destination';
   options: ILocalFileDestinationProviderOptions;
+  #providersMetadata: { source?: IMetadata; destination?: IMetadata } = {};
 
   constructor(options: ILocalFileDestinationProviderOptions) {
     this.options = options;
   }
 
-  #getDataTransformers() {
+  setMetadata(target: ProviderType, metadata: IMetadata): IDestinationProvider {
+    this.#providersMetadata[target] = metadata;
+
+    return this;
+  }
+
+  #getDataTransformers(options: { jsonl?: boolean } = {}) {
+    const { jsonl = true } = options;
```

Comment:
Is there a reason we want to disable jsonl for some files? It might be better to use it everywhere for consistency, even if we're not streaming it.

Reply:
You mean using it even for the metadata?
In theory we could, but I feel like jsonl is not meant to hold single json objects like the metadata

Reply:
It is weird to use it for a single line. But it's also weird to mix formats. That's why I'm not so sure. But I'm ok with it either way.